File: tests/fixtures/edgecase_repo/python/pkg/module4.py
"""Module 4 of the generated Python layer."""

from __future__ import annotations

from pkg.module3 import Service3


class Base4:
    """Base class for service 4."""


class Service4(Base4):
    """Service 4."""

    def handle(self, payload):
        """Handle a payload."""
        self.validate(payload)
        return Service3().handle(payload)

    def validate(self, payload):
        """Validate a payload."""
        return payload is not None


def helper_4(value):
    """Module-level helper."""
    return value

File: tests/fixtures/edgecase_repo/lua/module4.lua
local mod3 = require("module3")

local Base4 = {}
Base4.__index = Base4

function Base4.new(name)
  local self = setmetatable({}, Base4)
  self.name = name
  return self
end

local Service4 = setmetatable({}, { __index = Base4 })

function Service4:handle(payload)
  self:validate(payload)
  return true
end

function Service4:validate(payload)
  return payload ~= nil
end

local function helper4(value)
  return value
end

return Service4
